### PR TITLE
Routing integration

### PR DESCRIPTION
**Pull request by novemberborn**
Implements #71.

See <https://github.com/dojo/examples/pull/69> and <https://github.com/dojo/examples/pull/83> for an example.

* [x] Need to write tests
* [x] Need to update docs

### CODE REVIEW

**Comment by novemberborn** [comment]
See <https://github.com/dojo/examples/pull/69> for an example.

File: src/createApp.ts [hunk 1]
```diff
@@ -940,6 +1019,50 @@ const createApp = compose({
...
+					return {
+						pause() {},
+						resume() {},
+						destroy() {
+							realizationHandle.destroy();
+						}
+					};
+				}
+
+				const options = {} as RouterStartOptions;
```
Comment (kitsonk) [inline src/createApp.ts]: We aren't clear on this from a styleguide perspective, but I find:

```typescript
const options: RouterStartOptions = {};
```

To be more clear.  Also, would this be a good candidate for `Object.create(null);`?

Comment (novemberborn) [inline src/createApp.ts]: Nah the problem was that `RouterStartOptions` requires `dispatchCurrent` to be set. To use the default the options simply shouldn't be provided. So this cast was doing the wrong fix.

**Comment by novemberborn** [comment]
Now with tests and documentation, so ready for review!

File: README.md [hunk 2]
```diff
@@ -533,6 +556,27 @@ The realized DOM will be:
 </body>
 ```
 
+### Starting the application
+
+Use `App#start()` to start the application. If you provide the `root` option the `root` element is realized first. If a router was defined it is automatically started upon realization. The `dispatchCurrent` option is forwarded to `Router#start()`.
```
Comment (novemberborn) [inline README.md]: Need to document the returned handle.

File: src/createApp.ts [hunk 3]
```diff
@@ -949,6 +1028,49 @@ const createApp = compose({
 					};
 				});
 			});
+	},
+
+	start(this: App, { afterRealize, dispatchCurrent, root }: StartOptions = {}): Promise<PausableHandle> {
```
Comment (novemberborn) [inline src/createApp.ts]: Should ensure `start()` can only be called once.

**Comment by novemberborn** [comment]
Addressed my own comments.

**Pull request merged**